common/src/main/java/org/keycloak/common/util/PemUtils.java

```diff
@@ -157,4 +158,7 @@ public static String removeBeginEnd(String pem) {
         return pem.trim();
     }
 
+    public static String generateThumbprint(String[] certChain, String encoding) throws NoSuchAlgorithmException, IOException {
+        return Base64Url.encode(pemToDer(certChain[0]));
```

Comment:
This looks strange to me as there is no thumbprint involved at all? It just directly uses whole certificate?

Reply:
Ah, whoops, I boogered it.  Left out the most important part: `MessageDigest.getInstance(encoding).digest(...)`.  Fix coming shortly.